### DISCUSSION TITLE
display a object in a .isml

### DISCUSSION THREAD

**Discussion by claethem** [SalesforceCommerceCloud/sfcc-ci#543] [Team Posts]
Hi,

in my template (.isml) i want display all the pamameter of my object.

actualy i try

`${product}`

i get 

`[object Object]`

there is a possibility to see all the attributes of this object? 

Best regards.

**Comment by nimitshukla123** [comment]
Use JSON.stringify to convert object into string and then you will be able
to see complete object information.

On Thu, 25 Apr 2019 at 4:43 AM, claethem <notifications@github.com> wrote:

> Hi,
>
> in my template (.isml) i want display all the pamameter of my object.
>
> actualy i try
>
> ${product.variationAttributes[0].values}
>
> i get:
>
> [Ljava.lang.Object;@19d1c30
>
> there is a possibility to see all the attributes of this object?
>
> Best regards.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/31>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADMLMXYPZKRP4ST233X6ZELPSDBAJANCNFSM4HIHXHJA>
> .
>
-- 
Thanks & Regards,

Nimit Shukla
Mobile: +65 84575208

**Comment by zlatinz** [comment]
JSON.stringify(Object.keys(theObject)) will give you an array of the name of the attributes you have in theObject.
Please note that JSON.stringify does not work with Script API framework objects. It will not process them due to recursive relations which could exist (e.g. product object having recommendations collection of product objects, which have recommendations collection, etc. which can in some cases result in having to load the whole catalog in memory to render the result JSON.)

You can also use Array.prototype.slice.call(aCollection) to convert aCollection (list of values) to a JavaScript Array, which you can pass as argument to JSON.stringify()
